### PR TITLE
More frontend work

### PR DESCRIPTION
**Pull request by inglesp**
This batches up several changes, and should be reviewed commit-by-commit.

The main change is that `CodelistBuilder` now uses the new `TreeTables` component, which means that large codelists can be rendered by React, and are more tractable for users.

This change also allows a couple of other changes in behaviour, including showing slightly more information in the modal popups.

There'll be another PR to delete a bunch of unused code.

### CODE REVIEW

File: static/src/js/tree/index.jsx [hunk 1]
```diff
@@ -1,26 +1,37 @@
...
   readValueFromPage("parent-map"),
   readValueFromPage("child-map")
 );
 
+const treeTables = readValueFromPage("tree-tables");
+const codeToStatus = readValueFromPage("code-to-status");
+const codeToTerm = readValueFromPage("code-to-term");
+
+const ancestorCodes = treeTables
+  .map(([_, ancestorCodes]) => ancestorCodes) // eslint-disable-line no-unused-vars
```
Comment (ghickman) [inline static/src/js/tree/index.jsx:20]: Does JS not have a way to ignore variables when looping?? (arg!)

File: static/src/js/builder/index.jsx [hunk 2]
```diff
@@ -21,7 +21,8 @@ const ancestorCodes = treeTables
   .flat();
 const visiblePaths = hierarchy.initiallyVisiblePaths(
   ancestorCodes,
-  codeToStatus
+  codeToStatus,
+  1
```
Comment (ghickman) [inline static/src/js/builder/index.jsx:25]: Is there a way to make this a kwarg/named argument in JS?  (without going the full on Object route at least)

**Review by ghickman — APPROVED** [review]

File: static/src/js/tree/index.jsx (hunk 1 above)
Comment (inglesp) [reply static/src/js/tree/index.jsx]: Turns out there is!  Fixed in 2405a22f.

File: static/src/js/builder/index.jsx (hunk 2 above)
Comment (inglesp) [reply static/src/js/builder/index.jsx:25]: Not according to https://stackoverflow.com/questions/11796093/is-there-a-way-to-provide-named-parameters-in-a-function-call-in-javascript.

Agree, it's unsightly.

**Pull request merged**